strongbox-event-api/src/main/java/org/carlspring/strongbox/event/AsyncEventListener.java

```diff
@@ -6,13 +6,14 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-import org.springframework.context.event.EventListener;
 import org.springframework.scheduling.annotation.Async;
+import org.springframework.transaction.event.TransactionPhase;
+import org.springframework.transaction.event.TransactionalEventListener;
 
 @Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
-@EventListener
+@TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
 @Async("eventTaskExecutor")
 public @interface AsyncEventListener
```

Comment:
Name `AsyncEventListener`:
Right now it's now very clear if I use `AsyncEventListener` that it's also related with some `AFTER_COMMIT` transaction state. I mean,  `AsyncEventListener` name may also mean that I can use it just as an async listener without any transactional consideration.

Consider dropping or renaming this class, since `@TransactionalEventListener` may work with other transational phases too.